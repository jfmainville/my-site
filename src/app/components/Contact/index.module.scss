@import "../../variables.module.scss";

.Contact {
  display: flex;
  flex-direction: column;
  background-color: $secondary-color;
}

.ContactSection {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
}

@media (max-width: $small-screen) {
  .ContactSection {
    flex-direction: column;
  }
}

.ContactSectionHeader {
  display: flex;
  flex-wrap: wrap;
}

.ContactSectionForm {
  display: flex;
  flex-direction: row;
}

.Input {
  display: flex;
  flex-direction: row;
  outline: none;
  box-shadow: none;
  border: 1px solid;
  height: 2rem;
  width: 15rem;
  margin: 0.5rem 0 0.5rem 0;
  background-color: $secondary-color;
}

.TextArea {
  display: flex;
  flex-direction: row;
  outline: none;
  box-shadow: none;
  border: 1px solid;
  height: 10rem;
  width: 30rem;
  margin: 0.5rem 0 0.5rem 0;
  background-color: $secondary-color;
}

.SubmitButton {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 15rem;
  margin: 2rem 0 2rem 0;
  cursor: pointer;
  outline: none;
  border: 1px solid $secondary-color;
  color: $secondary-color;
  height: 2rem;
  background-color: $text-color;
}

.TextHeading {
  margin: 2rem;
}
