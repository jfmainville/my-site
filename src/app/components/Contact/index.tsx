import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "./index.module.scss";
import { FormEvent } from "react";
import Link from "next/link";
import {
  faFacebook,
  faGithub,
  faLinkedin,
  faXTwitter,
} from "@fortawesome/free-brands-svg-icons";
import { faX } from "@fortawesome/free-solid-svg-icons";

type Props = {
  handleSendEmail: (event: FormEvent<HTMLFormElement>) => void;
};

const Contact = ({ handleSendEmail }: Props) => {
  return (
    <section id={"contact"} className={styles.Contact}>
      <h1 className={styles.TextHeading}>Contact</h1>
      <div className={styles.ContactSection}>
        <div className={styles.ContactSectionHeader}>
          <h2 className={styles.TextHeading}>
            Looking forward to hearing from you!
          </h2>{" "}
          <div className={styles.ContactSocialMedia}>
            <div className={styles.ContactSocialMediaEntry}>
              <Link href="https://github.com/jfmainville">
                <FontAwesomeIcon icon={faGithub} size="2x" />
              </Link>
            </div>
            <div className={styles.ContactSocialMediaEntry}>
              <Link href="https://www.linkedin.com/in/jean-frederic-mainville-8566b485">
                <FontAwesomeIcon icon={faLinkedin} size="2x" />
              </Link>
            </div>
            <div className={styles.ContactSocialMediaEntry}>
              <Link href="https://www.facebook.com/jeanfrederic.mainville/">
                <FontAwesomeIcon icon={faFacebook} size="2x" />
              </Link>
            </div>
            <div className={styles.ContactSocialMediaEntry}>
              <Link href="https://x.com/JeanMainvi">
                <FontAwesomeIcon icon={faXTwitter} size="2x" />
              </Link>
            </div>
          </div>
        </div>
        <div className={styles.ContactSectionForm}>
          <form id="SubmitForm" onSubmit={handleSendEmail}>
            <label htmlFor="firstName" className={styles.InputLabel}>
              First Name
            </label>
            <input
              className={styles.Input}
              id="firstName"
              name="firstName"
              required={true}
              type="text"
            />
            <label htmlFor="lastName" className={styles.InputLabel}>
              Last Name
            </label>
            <input
              className={styles.Input}
              id="lastName"
              name="lastName"
              required={true}
              type="text"
            />
            <label htmlFor="email" className={styles.InputLabel}>
              Email
            </label>
            <input
              className={styles.Input}
              id="email"
              name="email"
              required={true}
              type="email"
            />
            <label htmlFor="phone" className={styles.InputLabel}>
              Phone
            </label>
            <input
              className={styles.Input}
              id="phone"
              name="phone"
              type="text"
            />
            <label htmlFor="message" className={styles.InputLabel}>
              Message
            </label>
            <textarea
              className={styles.TextArea}
              id="message"
              name="message"
              required={true}
            />
            <button className={styles.SubmitButton} type="submit">
              Submit
            </button>
          </form>
        </div>
      </div>
    </section>
  );
};

export default Contact;
