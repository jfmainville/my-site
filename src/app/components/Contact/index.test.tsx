import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import Contact from "./index";

describe("<Contact/> Test Suite", () => {
  it("should render the <Contact/> component unchanged", () => {
    const handleSendEmail = jest.fn();
    const { container } = render(<Contact handleSendEmail={handleSendEmail} />);
    expect(container).toMatchSnapshot();
  });

  it("should contain the Contact heading in the <Contact/> component", () => {
    const handleSendEmail = jest.fn();
    render(<Contact handleSendEmail={handleSendEmail} />);
    const heading = screen.getByText("Contact");
    expect(heading).toBeInTheDocument();
  });

  it("should contain all the required input fields in the <Contact/> component", () => {
    const handleSendEmail = jest.fn();
    render(<Contact handleSendEmail={handleSendEmail} />);
    const firstName = screen.getByLabelText("First Name");
    const lastName = screen.getByLabelText("Last Name");
    const email = screen.getByLabelText("Email");
    const phone = screen.getByLabelText("Phone");
    const message = screen.getByLabelText("Message");
    expect(firstName).toBeInTheDocument();
    expect(lastName).toBeInTheDocument();
    expect(email).toBeInTheDocument();
    expect(phone).toBeInTheDocument();
    expect(message).toBeInTheDocument();
  });

  it("should fill the all the required form input fields in the <Contact/> component", () => {
    const handleSendEmail = jest.fn();
    render(<Contact handleSendEmail={handleSendEmail} />);
    const firstName = screen.getByRole("textbox", { name: "First Name" });
    const lastName = screen.getByRole("textbox", { name: "Last Name" });
    const email = screen.getByRole("textbox", { name: "Email" });
    const phone = screen.getByRole("textbox", { name: "Phone" });
    const message = screen.getByRole("textbox", { name: "Message" });
    fireEvent.change(firstName, { target: { value: "John" } });
    fireEvent.change(lastName, { target: { value: "Doe" } });
    fireEvent.change(email, { target: { value: "jdoe@test.com" } });
    fireEvent.change(phone, { target: { value: "555-555-5555" } });
    fireEvent.change(message, { target: { value: "This is a test message." } });
    expect(firstName).toHaveValue("John");
    expect(lastName).toHaveValue("Doe");
    expect(email).toHaveValue("jdoe@test.com");
    expect(phone).toHaveValue("555-555-5555");
    expect(message).toHaveValue("This is a test message.");
  });

  it("should execute the sendEmail function in the <Contact/> component", async () => {
    const handleSendEmail = jest.fn((event) => event.preventDefault());
    const user = userEvent.setup();
    render(<Contact handleSendEmail={handleSendEmail} />);
    const firstName = screen.getByRole("textbox", { name: "First Name" });
    const lastName = screen.getByRole("textbox", { name: "Last Name" });
    const email = screen.getByRole("textbox", { name: "Email" });
    const phone = screen.getByRole("textbox", { name: "Phone" });
    const message = screen.getByRole("textbox", { name: "Message" });
    fireEvent.change(firstName, { target: { value: "John" } });
    fireEvent.change(lastName, { target: { value: "Doe" } });
    fireEvent.change(email, { target: { value: "jdoe@test.com" } });
    fireEvent.change(phone, { target: { value: "555-555-5555" } });
    fireEvent.change(message, { target: { value: "This is a test message." } });

    const submit = screen.getByRole("button", { name: "Submit" });
    await user.click(submit);
    expect(handleSendEmail).toHaveBeenCalled();
  });
});
