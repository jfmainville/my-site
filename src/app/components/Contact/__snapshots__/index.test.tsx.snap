// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact/> Test Suite should render the <Contact/> component unchanged 1`] = `
<div>
  <section
    class="Contact"
    id="contact"
  >
    <h1
      class="TextHeading"
    >
      Contact
    </h1>
    <div
      class="ContactSection"
    >
      <div
        class="ContactSectionHeader"
      >
        <h2
          class="TextHeading"
        >
          Looking forward to hearing from you!
        </h2>
         
      </div>
      <div
        class="ContactSectionForm"
      >
        <form
          id="SubmitForm"
        >
          <label
            for="firstName"
          >
            First Name
          </label>
          <input
            class="Input"
            id="firstName"
            name="firstName"
            required=""
            type="text"
          />
          <label
            for="lastName"
          >
            Last Name
          </label>
          <input
            class="Input"
            id="lastName"
            name="lastName"
            required=""
            type="text"
          />
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="Input"
            id="email"
            name="email"
            required=""
            type="email"
          />
          <label
            for="phone"
          >
            Phone
          </label>
          <input
            class="Input"
            id="phone"
            name="phone"
            type="text"
          />
          <label
            for="message"
          >
            Message
          </label>
          <textarea
            class="TextArea"
            id="message"
            name="message"
            required=""
          />
          <button
            class="SubmitButton"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </section>
</div>
`;
